qemu/tests/x86_cpu_v_spec_ctrl.py

```diff
@@ -0,0 +1,75 @@
+import os
+import logging
+
+from avocado.core import exceptions
+
+from virttest import cpu
+from virttest import data_dir
+from virttest import env_process
+from virttest import error_context
+from virttest.utils_test import update_boot_option
+
+from provider.cpu_utils import check_cpu_flags
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    support Virtual SPEC_CTRL inside guest
+
+    1. check the 'v_spec_ctrl' on supported host
+    2. add guest kernel command line 'spec_store_bypass_disable=on'
+    3. verify the guest sets the spec ctrl properly on all the cpus.
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+    flags = params["flags"]
+    check_host_flags = params.get_boolean("check_host_flags")
+    if check_host_flags:
+        check_cpu_flags(params, flags, test)
+
+    supported_models = params.get("supported_models", "")
+    cpu_model = params.get("cpu_model")
+    if not cpu_model:
+        cpu_model = cpu.get_qemu_best_cpu_model(params)
+    if cpu_model not in supported_models.split():
+        test.cancel("'%s' doesn't support this test case" % cpu_model)
+
+    params["start_vm"] = "yes"
+    vm_name = params['main_vm']
+    env_process.preprocess_vm(test, params, env, vm_name)
+
+    proc_cmdline = params["proc_cmdline"]
+    vm = env.get_vm(vm_name)
+    session = vm.wait_for_login()
+    boot_option = params["boot_option"]
+    check_output = str(session.cmd(proc_cmdline, timeout=60)).split()
+    if boot_option and boot_option not in check_output:
```

Comment:
This step is already included in update_boot_option().

Reply:
> This step is already included in update_boot_option().

Check the kernel args early and also can use this to reset the default kernel cmdline.